File: core/src/main/java/com/graphhopper/routing/ev/Hazmat.java
package com.graphhopper.routing.ev;

import com.graphhopper.util.Helper;

/**
 * Defines general restrictions for the transport of hazardous materials.<br>
 * If not tagged it will be {@link #YES}
 */
public enum Hazmat {
    YES, NO;

    public static final String KEY = "hazmat";

    public static EnumEncodedValue<Hazmat> create() {
        return new EnumEncodedValue<>(KEY, Hazmat.class);
    }
}

File: core/src/main/kotlin/com/graphhopper/routing/ev/Hazmat.kt
package com.graphhopper.routing.ev

import com.graphhopper.util.Helper

/**
 * Defines general restrictions for the transport of hazardous materials.<br></br>
 * If not tagged it will be [YES]
 */
enum class Hazmat {
    YES, NO;

    override fun toString(): String = Helper.toLowerCase(super.toString())

    companion object {
        const val KEY = "hazmat"

        @JvmStatic
        fun create(): EnumEncodedValue<Hazmat> = EnumEncodedValue(KEY, Hazmat::class.java)
    }
}
